TodoWithControllersAuthJWT/TodoController.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Todos
+{
+    [ApiController]
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]
+    [Route("/api/todos")]
+    public class TodoController : ControllerBase
+    {
+        private readonly TodoDbContext _db;
+        public TodoController(TodoDbContext db)
+        {
+            _db = db ?? throw new ArgumentNullException(nameof(db));
+        }
+        [AllowAnonymous]
+        [HttpGet(nameof(GenerateToken))]
+        public IActionResult GenerateToken([FromQuery] string username, [FromQuery] string password, [FromQuery] string[] desiredClaims)
```

Comment:
I think token generation should be done in a separate controller/route action etc.